keyboard/src/modules/Keyboard.rb

```diff
@@ -1504,6 +1483,43 @@ def write_udev_rule
     end
   end
 
+  # Checks if the graphical environment is being executed remotely using
+  # "ssh -X"
+  def x11_over_ssh?
+    display = Builtins.getenv("DISPLAY") || ""
+    display.split(":")[1].to_i >= 10
```

Comment:
What about `display[1..-1].to_i` ? AFAIK when using forwarded X there is no hostname in `DISPLAY` in front of `:`...